app/means_test/payload.py

```diff
@@ -0,0 +1,515 @@
+                "income_support": "income_support" in form_data.get("benefits", []),
+            }
+            if session.get_eligibility().on_benefits
+            else {}
+        )
+
+        payload = {
+            "specific_benefits": benefits,
+            "on_passported_benefits": is_passported,
+        }
+
+        if is_passported:
+            payload = recursive_update(payload, IncomePayload.default())
+            payload = recursive_update(payload, OutgoingsPayload.default())
+        else:
+
+            def val(field):
+                return form_data.get(field)
+
+            payload["you"] = {
+                "income": {"child_benefits": MoneyInterval(val("child_benefit"))}
+            }
+
+        self.update(payload)
+
+
+class AdditionalBenefitsPayload(dict):
+    def __init__(self, form_data={}):
+        super(AdditionalBenefitsPayload, self).__init__()
+
+        def val(field):
+            return form_data.get(field)
+
+        def yes(field):
+            return form_data[field] is True
+
+        benefits = val("benefits")
+
+        payload = {
+            "on_nass_benefits": False,
+            "you": {
+                "income": {
+                    "benefits": MoneyInterval(val("total_other_benefit"))
+                    if yes("other_benefits")
+                    else MoneyInterval(0)
+                }
+            },
+        }
+
+        if benefits:
+            payload["notes"] = "Other benefits:\n - {0}".format("\n - ".join(benefits))
+
+        self.update(payload)
+
+
+class PropertyPayload(dict):
+    def __init__(self, form_data={}):
+        super(PropertyPayload, self).__init__()
+
+        def val(field):
+            return form_data.get(field)
+
+        def yes(field):
+            return form_data.get(field) is True
+
+        def no(field):
+            return form_data.get(field) is False
+
+        self.update(
+            {
+                "value": to_amount(val("property_value")),
+                "mortgage_left": to_amount(val("mortgage_remaining")),
+                "share": 100 if no("other_shareholders") else None,
+                "disputed": val("in_dispute"),
+                "rent": MoneyInterval(val("rent_amount"))
+                if yes("is_rented")
+                else MoneyInterval(0),
+                "main": val("is_main_home"),
+            }
+        )
+
+
+class PropertiesPayload(dict):
+    @classmethod
+    def default(cls):
+        return {
+            "property_set": [],
+            "you": {
+                "deductions": {"mortgage": MoneyInterval(0)},
+                "income": {"other_income": MoneyInterval(0)},
+            },
+        }
+
+    def __init__(self, form_data={}):
+        super(PropertiesPayload, self).__init__()
+
+        properties = [
+            PropertyPayload(prop)
+            for prop in form_data.get("properties", [])
+            if session.get_eligibility().owns_property
```

Comment:
This needs to be read/executed reverse. So order of reading is line 201, 200 and 199.

What do you think of breaking this down into a normal for loop wrapped in a if condition?